packages/dd-trace/src/plugins/util/web.js

```diff
@@ -361,6 +364,32 @@ const web = {
     const res = context.res
 
     scope.bind(res, context.span)
+  },
+  obfuscateQs (obfuscator, url) {
+    if (obfuscator === false) return url
+
+    const i = url.indexOf('?')
+
+    if (i === -1) return url
+
+    const path = url.slice(0, i)
+
+    if (obfuscator === true) return path
+
+    let qs = url.slice(i + 1)
+
+    try {
+      qs = vm.runInNewContext('decodeURIComponent(qs).replace(obfuscator, \'<redacted>\')', {
```

Comment:
This needs a synthetic benchmark as well as a real world benchmark (sirun) since it can easily happen thousands of times per second. The timeout is also not enough a limit for the same reason. What happens if 1000 requests are blocked for 200ms?

Reply:
What value would the benchmarks bring ? We know it is blocking already and we know for how much time max no ? I don't see the information a benchmark would add.

Reply:
We've decided to reduce the timeout to 5ms